source/common/config/grpc_mux_impl.h

```diff
@@ -39,6 +39,16 @@ class GrpcMuxImpl : public GrpcMux,
               Random::RandomGenerator& random, Stats::Scope& scope,
               const RateLimitSettings& rate_limit_settings, bool skip_subsequent_node);
 
+  ~GrpcMuxImpl() override;
+
+  // Causes all GrpcMuxImpl objects to stop sending any messages on `grpc_stream_` to fix a crash
+  // on Envoy shutdown due to dangling pointers. This may not be the ideal fix; it is probably
```

Comment:
Can you add more details about the crash (potentially in the tracking issue referenced here in a TODO) so we don't lose all your debugging context? It's not immediately clear to me how your other change caused this crash to show up, so we should capture that somehow.

Reply:
I'll add a link to the issue (#15072). I put some notes in there earlier.

I'm not exactly sure what in this PR caused that issue to go from flaky-reproducible to 100% reproducible, but I think something is deleted in a different order, probably due to deferredDelete.